### PR TITLE
Feat/gps backend

### PR DESCRIPTION
**Pull request by laurawheatley**
Implementing backend for GPS filter [#95], refers to [#96].

Written with @anthonyle17.

### CODE REVIEW

**Comment by lkadian** [comment]
I can't really get any location stamp to appear on my picture after they're taken, is it normal? I enabled it in the settings and I also enabled location permissions for this app

File: app/src/main/res/layout/activity_main.xml [hunk 1]
```diff
@@ -18,6 +18,27 @@
         android:layout_marginTop="0dp"
         android:layout_marginEnd="0dp" />
 
+    <LinearLayout
```
Comment (lkadian) [inline app/src/main/res/layout/activity_main.xml:21]: Is this being used? Or are we directly transforming the bitmap instead?

**Comment by lkadian** [comment]
So the GPS tag functionality itself works great, however there are some things we need to fix

   * I discussed this with @laurawheatley , the placement of the coordinates printed on the final picture are hard coded, so on my phone I couldn't even see them appear, I had to set the X value in `canvas.drawText(addressCoordina...)` to 300F to have them show up, which is a lot different than the one hard coded now. I think this is being looked into right now. Maybe get the size of the picture/bitmap generated and adjust placement accordingly

   * I'm not sure "Photo location stamp" in the settings is doing anything, right now the coordinates are always being put on my picture whether its disabled or not

**Comment by anthonyle17** [comment]
> So the GPS tag functionality itself works great, however there are some things we need to fix
> 
> * I discussed this with @laurawheatley , the placement of the coordinates printed on the final picture are hard coded, so on my phone I couldn't even see them appear, I had to set the X value in `canvas.drawText(addressCoordina...)` to 300F to have them show up, which is a lot different than the one hard coded now. I think this is being looked into right now. Maybe get the size of the picture/bitmap generated and adjust placement accordingly
> * I'm not sure "Photo location stamp" in the settings is doing anything, right now the coordinates are always being put on my picture whether its disabled or not

Everything you mentioned was fixed and pushed into the PR except one thing:
-Cannot get the size of the text of the address location so instead of stamping the location on the top right of a picture, I decided to stamp the location on the top left of a picture. The reason behind this is because some addresses can be very long, hence it may not be displayed if we start printing the address at a specific F point close to the right end of the bitmap. If we start printing the address on the left side of the bitmap, there will be more space for the address; hence, higher chance of displaying the whole address.

**Comment by lkadian** [comment]
Great job @laurawheatley @anthonyle17 , it works great on my phone now. I will fix the conflict and merge

One minor improvement we could do later on if we have time is have the system prompt whether or not the user wants to allow location permissions when he clicks "enable GPS tagging" in the settings. Right now it wasn't asking so I had to go to the settings and do it myself. Just a minor thing we can consider adding later, but it's not a huge deal

**Pull request merged**